File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        // To store the result prefix
        StringBuilder ans = new StringBuilder();

        // Sort the array of strings
        Arrays.sort(strs);

        // First string after sorting
        String first = strs[0];

        // Last string after sorting
        String last = strs[strs.length - 1];

        // Compare characters of the first and last strings
        for (int i = 0; i < Math.min(first.length(), last.length()); i++) {
            // Stop if characters are different
            if (first.charAt(i) != last.charAt(i)) {
                return ans.toString();
            }

            // Add matching character to result
            ans.append(first.charAt(i));
        }

        // Return the final common prefix
        return ans.toString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        
        // Step 1: Sort the strings alphabetically
        sort(strs.begin(), strs.end());
        
        // Step 2: Compare the first and last strings
        string first = strs.front();
        string last = strs.back();
        string result = "";
        
        // Match characters sequentially
        for (int i = 0; i < min(first.length(), last.length()); i++) {
            if (first[i] != last[i]) {
                break; // Mismatch found, stop comparing
            }
            result += first[i];
        }
        
        return result;
    }
};